coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -550,15 +557,12 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
     // get a mapping of assignments to time-ranges to query
     val lookbackMs = getLookBackMillis(logicalPlan).max
     val offsetMs = getOffsetMillis(logicalPlan).max
-    val assignmentRanges = {
-      // "distinct" in case this is a BinaryJoin
-      val partitions = getPartitions(logicalPlan, qParams).distinct.sortBy(_.timeRange.startMs)
-      require(!partitions.isEmpty, s"Partition assignments is not expected to be empty for query ${qParams.promQl}")
-      val timeRange = TimeRange(1000 * qParams.startSecs, 1000 * qParams.endSecs)
-      val stepMsOpt = if (qParams.startSecs == qParams.endSecs) None else Some(1000 * qParams.stepSecs)
-      getAssignmentQueryRanges(partitions, timeRange,
+    val timeRange = TimeRange(1000 * qParams.startSecs, 1000 * qParams.endSecs)
+    val stepMsOpt = if (qParams.startSecs == qParams.endSecs) None else Some(1000 * qParams.stepSecs)
+    val partitions = getPartitions(logicalPlan, qParams).distinct.sortBy(_.timeRange.startMs)
+    require(partitions.nonEmpty, s"Partition assignments is not expected to be empty for query ${qParams.promQl}")
```

Comment:
Confirming: this won't affect, say, scalar queries?

Reply:
Great point, this forced me to try few cases which failed with this. I pushed a fix for that. As far as the above line is concerned we are good.

```
Caused by: java.lang.IllegalArgumentException: requirement failed
	at scala.Predef$.require(Predef.scala:268)
	at filodb.query.BinaryJoin.<init>(LogicalPlan.scala:462)
	at filodb.query.BinaryJoin.copy(LogicalPlan.scala:460)
	at filodb.coordinator.queryplanner.PlannerUtil$.rewritePlanWithRemoteRawExport(DefaultPlanner.scala:666)
```